tooling/src/hypothesistooling/releasemanagement.py

```diff
@@ -0,0 +1,154 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis-python
+#
+# Most of this work is copyright (C) 2013-2018 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+"""Helpful common code for release management tasks that is shared across
+multiple projects.
+
+Note that most code in here is brittle and specific to our build and
+probably makes all sorts of undocumented assumptions, even as it looks
+like a nice tidy reusable set of functionality.
+"""
+
+
+from __future__ import division, print_function, absolute_import
+
+import re
+from datetime import datetime, timedelta
+
+
+def release_date_string():
+    """Returns a date string that represents what should be considered "today"
+    for the purposes of releasing. It is always measured in UTC, but if it's in
+    the last hour of the day it will actually be considered tomorrow.
+
+    The reason for counting it as the later day is that it ensures that
+    (unless our release process takes more than 23 hours) this value
+    remains consistent throughout the entire release.
+    """
+    now = datetime.utcnow()
+
+    return max([
+        d.strftime('%Y-%m-%d') for d in (now, now + timedelta(hours=1))
+    ])
+
+
+def replace_assignment_in_string(contents, name, value):
+    lines = contents.split('\n')
+
+    matcher = re.compile(r'\A(\s*%s\s*=\s*)' % (name,))
```

Comment:
This looks new to me. An example of matched output would be helpful.

Reply:
It is new. I'll add one.